lib/commands/cache.js

```diff
@@ -298,18 +298,23 @@ class Cache extends BaseCommand {
       if (!this.npm.config.get('force')) {
         throw this.usageError('Please use --force to remove entire npx cache')
       }
+      const { npxCache } = this.npm.flatOptions
+      if (!this.npm.config.get('dry-run')) {
+        return fs.rm(npxCache, { recursive: true, force: true })
+      }
     }
+
     const cache = await this.#npxCache(keys)
     for (const key in cache) {
       const { path: cachePath } = cache[key]
       output.standard(`Removing npx key at ${cachePath}`)
       if (!this.npm.config.get('dry-run')) {
-        return fs.rm(cachePath, { recursive: true })
+        await fs.rm(cachePath, { recursive: true })
```

Comment:
@wraithgar I don't think we wanted `return` here, or we'd just remove the first key in the cache

Reply:
You're right, oops.  Hopefully a test removing multiple keys would have exposed this bug.

Reply:
That's how I found it 😂